jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3219,9 +3229,12 @@ private static final boolean hasMetaChars(String s) {
 	 * @since 1.4
 	 */
 	public String[] split(String regex, int max) {
-		// it is faster to handle simple splits inline (i.e. no fancy regex matching)
+		// it is faster to handle simple splits inline (i.e. no fancy regex matching),
+		// including single escaped literal character (e.g. \. \{),
 		// so we test for a suitable string and handle this here if we can
-		if (regex != null && regex.lengthInternal() > 0 && !hasMetaChars(regex)) {
+		boolean singleEscapeLiteral = isSingleEscapeLiteral(regex);
+		if ((regex != null) && (regex.lengthInternal() > 0) 
+				&& (!hasMetaChars(regex) || singleEscapeLiteral)) {
```

Comment:
The formatting here is off.  Can you either do this in 1 long if statement, or remove one of the tabs from the second line, or (my favourite), format like:
```
		if ((regex != null) && (regex.lengthInternal() > 0) 
 		&& (!hasMetaChars(regex) || singleEscapeLiteral)) 
 		{
```